### PR TITLE
[4.0] Add prepared statements for plg_authentication_joomla

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions

**This PR affect the Joomla Authentication System - handle with care**

Use the plugin in all ways you can think of.

* Login
* Logout
* close/open browser
* user different users

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/authentication/joomla/joomla.php [hunk 1]
```diff
@@ -67,7 +67,8 @@ public function onUserAuthenticate($credentials, $options, &$response)
 		$query = $this->db->getQuery(true)
 			->select('id, password')
```
Comment (Quy) [inline plugins/authentication/joomla/joomla.php]: Apply `$db->quoteName`

File: plugins/authentication/joomla/joomla.php [hunk 2]
```diff
@@ -67,7 +67,8 @@ public function onUserAuthenticate($credentials, $options, &$response)
...
 			->from('#__users')
-			->where('username=' . $this->db->quote($credentials['username']));
+			->where('username = :username')
```
Comment (Quy) [inline plugins/authentication/joomla/joomla.php]: Apply `$db->quoteName`

File: plugins/authentication/joomla/joomla.php [hunk 3]
```diff
@@ -65,9 +65,10 @@ public function onUserAuthenticate($credentials, $options, &$response)
 		}
 
 		$query = $this->db->getQuery(true)
-			->select('id, password')
-			->from('#__users')
-			->where('username=' . $this->db->quote($credentials['username']));
+			->select($db->quoteName(['id', 'password']))
```
Comment (Quy) [inline plugins/authentication/joomla/joomla.php:69]: Should it be `$this->db->`?

Comment (HLeithner) [reply plugins/authentication/joomla/joomla.php:69]: yep or the other way around. thx

**Review by alikon — APPROVED** [review]

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**